### DISCUSSION TITLE
What's the difference between this and GPT4all

### DISCUSSION THREAD

**Discussion by gael-vanderlee** [imartinez/privateGPT#364] [General]
Hello,

I'm pretty new to this space.
GPT4all claims to run locally and to ingest documents as well. What are the differences with this project ? Any reason to pick one over the other ?

**Comment by dhirajsuvarna** [comment]
This is not a replacement of GPT4all, but rather uses it to achieve a specific task, i.e. querying over the documents using langchain framework. 
It does this by using GPT4all model, however, any model can be used and sentence_transformer embeddings, which can also be replaced by any embeddings that langchain supports. 

So see this project as an application project and not as a LLM project.

**Comment by gael-vanderlee** [comment]
I understand, thanks for the reponse!

**Comment by hcchengithub** [comment]
Gvanderl was not asking "what's the difference between privateGPT and GPT4All" which dhirajsuvarna has however nicely answered.
What actually asked was "what's the difference between privateGPT and GPT4All's plugin feature 'LocalDocs'"

If they are actually same thing I'd like to know.

I actually tried both, GPT4All is now v2.4.10 and it's LocalDocs plugin is confusing me. GPT4ALL answered query but I can't tell did it refer to LocalDocs or not. It sometimes list references of sources below it's anwer, sometimes not. This is the GPT4ALL UI's problem anyway. 

While privateGPT works fine.
